src/scheduler.c

```diff
@@ -2778,12 +2896,49 @@ struct task *scheduler_done(struct scheduler *s, struct task *t) {
   /* Mark the task as skip. */
   t->skip = 1;
 
+  t->done = 1;
+
   /* Return the next best task. Note that we currently do not
      implement anything that does this, as getting it to respect
      priorities is too tricky and currently unnecessary. */
   return NULL;
 }
 
+struct task *signal_sleeping_runners(struct scheduler *s, struct task *t) {
```

Comment:
What does this function do?

Reply:
For the GPU pack tasks we only unlock the cell and set t->skip=1 at the end of runner_main_clean.cu without unlocking their dependencies. I split scheduler_done() into a few functions (signal_sleeping_runners and enqueue_dependencies) which decrement the number of waiting tasks and enqueue the deps. I did this to play around with how and when things are done but we should now just be able to call scheduler_done() for each GPU task individually after we have unpacked the results